todo_app/tests/test_app.py

```diff
@@ -0,0 +1,47 @@
+import os
+import pytest
+import requests
+from dotenv import load_dotenv, find_dotenv
+from todo_app import app
+
+@pytest.fixture
+def client():
+    file_path = find_dotenv('.env.test')
+    load_dotenv(file_path, override=True)
+
+    test_app = app.create_app()
+
+    with test_app.test_client() as client:
+        yield client
+
+def test_index_page(monkeypatch, client):
+    monkeypatch.setattr(requests, 'get', get_lists_stub)
+    response = client.get('/')
+    
+    assert response.status_code == 200
+    assert 'To do card' in response.data.decode()
+
+def test_add(monkeypatch, client):
+# Replace call to requests.get(url) with our own function
+    monkeypatch.setattr(requests, 'get', get_lists_stub)
```

Comment:
For a more reliable test you might want to mock requests.post **as well** as requests.get.

Currently the app makes a real HTTP request during this test (that fails but the app doesn't care). Would be better to avoid that in case the app does look at the response, or so that your test could assert that the correct post request was made. 

You would just want to make sure that the mock post function looks and behaves correctly.